dive_core/command_hierarchy.h

```diff
@@ -92,6 +89,53 @@ class Topology
     uint64_t GetChildNodeIndex(uint64_t node_index, uint64_t child_index) const;
     uint64_t GetNextNodeIndex(uint64_t node_index) const;
 
+protected:
+    struct ChildrenInfo
+    {
+        uint64_t m_start_index = UINT64_MAX;
+        uint64_t m_num_children = 0;
+    };
+
+    // List of all children for all nodes.
+
+    // m_children_list contains the node_indexes of all the "normal" children throughout the entire
+    // command hierarchy for a specific TopologyType (e.g., kSubmitTopology or kAllEventTopology).
+    // It's a concatenated list of all direct children, ordered by the parent node they belong to.
+    // For example,
+    //  if node A has children(C1, C2) and node B has children(C3),
+    //  then m_children_list might look like[C1_index, C2_index, C3_index, ...]
+    //
+    // The m_node_children vector then contains ChildrenInfo structs for each parent node. Each
+    // ChildrenInfo struct has a m_start_index and m_num_children. These values tell you where in
+    // m_children_list to find the children for a specific parent node.
+    DiveVector<uint64_t> m_children_list;
+
+    // This is a per-node pointer to m_children_list.
```

Comment:
the comment needs to be updated? there seems to be no pointer involved?

Reply:
Updated the comment.

Reply:
Thanks